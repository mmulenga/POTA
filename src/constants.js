/**
 * This file contains all string constants for the project
 * Use this instead of hardcoding string values
 * HTML and one-time-use messages are an acceptable exception
 */

/* Conditions */
export const Condition = {
  /* Cardiovascular Diseases */
  AtrialFib: 'Atrial fibrillation / History of irregular heart beat',
  Defib: 'Defibrillator / Pacemaker',
  CAD: 'Coronary artery disease / Cardiac stent',
  CerebralDisease: 'History of cerebral vascular disease / TIA / Stroke',
  VHD: 'Valvular heart disease / Valve Replacement',
  HeartFail: 'Heart failure',
  PVD: 'Peripheral Vascular disease (PVD)',
  /* Pulmonary Diseases */
  PulmonaryVascular: 'History of pulmonary vascular disease',
  PulmDisease: 'Severe COPD, Home Oxygen, Pulmonary HTN',
  /* Other Diseases */
  Risk: 'Age ≥ 69 years old OR ≥ 2 risk factors (HTN, DM, OSA, CKD, BMI > 35)',
  Bleeding: 'Bleeding disorders',
  Anemia: 'History of anemia in past 12 months or ongoing blood loss',
  KidneyDisease: 'Chronic kidney disease / on dialysis',
  Diabetes: 'Diabetes',
  Malignancy: 'Present malignancy / surgery for malignancy',
  Hepatic: 'Hepatic disease',
  Adrenal: 'History of adrenal disease',
  Pituitary: 'History of pituitary disease',
  Endocrine: 'History of major systemic endocrine disease',
  Thyroid: 'Known thyroid disease or on thyroid supplementation',
  Digoxin: 'Digoxin',
  Lithium: 'Lithium',
  Diuretics: 'Diuretics',
  ACEI: 'ACE-I',
  ARB: 'ARB',
  NSAIDS: 'NSAIDS',
  Anticoagulant: 'Anticoagulants',
  Antiplatelet: 'Antiplatelet (ASA excluded)',
  Steroid: 'Systemic steroid use within 6 months',
};

/* Medical Examinations */
export const Exam = {
  GnS: 'Group and Screen',
  ECG: 'ECG',
  CBC: 'CBC',
  RenPanel: 'Renal Panel (Creat + Lytes)',
  PTTINR: 'PTT/INR',
  LFT: 'LFT',
  Gluc: 'Gluc',
  HbA1C: 'HbA1C',
  CXR: 'CXR',
  TSH: 'TSH',
};

/* Glossary Entries */
export const Glossary = {
  AtrialFib: 'No extra information.',
  Defib: 'No extra information.',
  CAD: 'No extra information.',
  CardiacStent: 'No extra information.',
  CerebralDisease: 'No extra information.',
  PulmonaryVascular: 'No extra information.',
  VHD: 'No extra information.',
  HeartFail: 'No extra information.',
  PVD: 'No extra information.',
  /* Pulmonary Diseases */
  PulmDisease: '• Spirometric values AND symptomatic assessment should be sought. \n' +
               '• Post Bronchodilator FEV1: 30 - 50% of predicted; FEV1/FVC < 0.70 \n' +
               '• History of exacerbations (including prior hospitalizations) should be recorded.\n' +
               '• Worsening air flow limitation, greater SOB, reduced exercise capacity, having an ' +
               'impact on quality of life, and ability to complete activities of daily living should be present. \n',
  /* Other Diseases */
  Risk: 'No extra information.',
  Bleeding: '• Hemophiliac. \n' +
            '• History of extensive investigation due to a bleeding episode. \n' +
            '• History of extensive bleeding needing transfusion in previous surgeries. \n' +
            '• Bruising easily more than usual. \n',
  Anemia: 'No extra information.',
  ActiveBleeding: 'No extra information.',
  KidneyDisease: 'No extra information.',
  Renal: '(AKI). \n' +
         '• Chronic kidney disease / on dialysis. \n' +
         '• Diabetes. \n' +
         '• Heart failure. \n' +
         '• Age ≥ 65. \n' +
         '• Liver disease. \n' +
         '• Use of drugs with nephrotoxic potential in the perioperative period (in particular NSAIDs after surgery). \n',
  Diabetes: 'No extra information.',
  Malignancy: 'No extra information.',
  Hepatic: '• Alcohol abuse (i.e., female: 14 drinks / week; male: 21 drinks / week). \n' +
           '• Jaundice due to liver disease. \n' +
           '• All forms of hepatitis. \n' +
           '• Cirrhosis; Hepatic cancer. \n',
  Adrenal: 'No extra information.',
  Pituitary: '• Acromegaly. \n' +
             '• Cushing’s disease. \n' +
             '• Sheehan syndrome. \n' +
             '• Growth hormone deficiency. \n' +
             '• Hyper/hypo pituitarism. \n',
  Endocrine: '(Excluding Diabetes) \n' +
             '• Parathyroid disease \n' +
             '• Addison’s disease \n' +
             '• Cushing’s syndrome \n' +
             '• Graves’ disease \n',
  Thyroid: 'No extra information.',
  /* Medications */
  Digoxin: 'No extra information.',
  Lithium: 'No extra information.',
  Diuretics: 'Furosemide; Hydrochlorothiazide; Acetazolamide; Spironolactone; Chlorthalidone',
  ACEI: 'Captopril; Enalapril; Ramipril; Trandolapril; Perindopril',
  ARB: 'Valsartan; Candesartan; Irbesartan; Telmisartan; Losartan; Olmesartan',
  NSAIDS: 'No extra information.',
  Anticoagulant: 'Warfarin; Heparins; Dabigatran; Apixaban; Rivaroxaban',
  Antiplatelet: 'Clopidogrel; Ticagrelor; Prasugrel; Aspirin/Dipyridamole (Aggrenox)',
  Steroid: 'Prednisone; Prednisolone; Methylprednisolone; Hydrocortisone; Dexamethasone; Oral Budesonide',
};
