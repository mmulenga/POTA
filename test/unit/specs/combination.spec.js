import { Condition, Exam } from '@/constants';
import { PatientExamsNeeded } from '@/PreopRecommendation';

let requiredExams;
let conditionalExams;
let result;

describe('Testing PatientExamsNeeded() with multiple non-conditional comorbidities', () => {
  it('returns correct set of preop exams for AtrialFib + Defib', () => {
    result = PatientExamsNeeded([Condition.AtrialFib, Condition.Defib]);
    requiredExams = [Exam.ECG];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for AtrialFib + Defib to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for CAD + Bleeding', () => {
    result = PatientExamsNeeded([Condition.CAD, Condition.Bleeding]);
    requiredExams = [Exam.ECG, Exam.GnS, Exam.CBC, Exam.PTTINR];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for CAD + Bleeding to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for PVD + Anemia', () => {
    result = PatientExamsNeeded([Condition.PVD, Condition.Anemia]);
    requiredExams = [Exam.ECG, Exam.RenPanel, Exam.CBC];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for PVD + Anemia to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for CardiacStent + Adrenal', () => {
    result = PatientExamsNeeded([Condition.CardiacStent, Condition.Adrenal]);
    requiredExams = [Exam.ECG, Exam.RenPanel];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for CardiacStent + Adrenal to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for CardiacStent + Adrenal', () => {
    result = PatientExamsNeeded([Condition.CardiacStent, Condition.Adrenal]);
    requiredExams = [Exam.ECG, Exam.RenPanel];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for CardiacStent + Adrenal to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for CerebralDiseaese + Lithium', () => {
    result = PatientExamsNeeded([Condition.CerebralDisease, Condition.Lithium]);
    requiredExams = [Exam.ECG, Exam.RenPanel];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for CerebralDiseaese + Lithium to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for Stroke + Digoxin', () => {
    result = PatientExamsNeeded([Condition.Stroke, Condition.Digoxin]);
    requiredExams = [Exam.ECG, Exam.RenPanel];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for Stroke + Digoxin to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for HeartFail + Hepatic', () => {
    result = PatientExamsNeeded([Condition.HeartFail, Condition.Hepatic]);
    requiredExams = [Exam.ECG, Exam.CBC, Exam.RenPanel, Exam.PTTINR, Exam.LFT];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for HeartFail + Hepatic to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for PulmDisease + Renal + Steroid', () => {
    result = PatientExamsNeeded([Condition.PulmDisease, Condition.Renal, Condition.Steroid]);
    requiredExams = [Exam.ECG, Exam.CBC, Exam.CXR, Exam.RenPanel, Exam.Gluc];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for PulmDisease + Renal + Steroid to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for PulmonaryVascular + Pituitary + Steroid', () => {
    result = PatientExamsNeeded([Condition.PulmonaryVascular,
      Condition.Pituitary, Condition.Steroid]);
    requiredExams = [Exam.ECG, Exam.RenPanel, Exam.Gluc];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for PulmonaryVascular + Pituitary + Steroid to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for TIA + Hepatic + Steroid', () => {
    result = PatientExamsNeeded([Condition.TIA, Condition.Hepatic, Condition.Steroid]);
    requiredExams = [Exam.ECG, Exam.CBC, Exam.RenPanel, Exam.PTTINR, Exam.LFT, Exam.Gluc];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for TIA + Hepatic + Steroid to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for Bleeding + Digoxin + ACEI', () => {
    result = PatientExamsNeeded([Condition.Bleeding, Condition.Digoxin, Condition.ACEI]);
    requiredExams = [Exam.GnS, Exam.CBC, Exam.PTTINR, Exam.ECG, Exam.RenPanel];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for Bleeding + Digoxin + ACEI', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for Bleeding + Digoxin + ACEI', () => {
    result = PatientExamsNeeded([Condition.Bleeding, Condition.Digoxin, Condition.ACEI]);
    requiredExams = [Exam.GnS, Exam.CBC, Exam.PTTINR, Exam.ECG, Exam.RenPanel];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for Bleeding + Digoxin + ACEI', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for Endocrine + Thyroid + NSAIDS', () => {
    result = PatientExamsNeeded([Condition.Endocrine, Condition.Thyroid, Condition.NSAIDS]);
    requiredExams = [Exam.RenPanel, Exam.TSH];
    conditionalExams = [];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual(conditionalExams);
  });
  it('set of preop exams for Endocrine + Thyroid + NSAIDS', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });
});

describe('Testing PatientExamsNeeded() with multiple conditional comorbidities', () => {
  it('returns correct set of preop exams for VHD + Diabetes', () => {
    result = PatientExamsNeeded([Condition.VHD, Condition.Diabetes]);
    requiredExams = [Exam.ECG, Exam.RenPanel, Exam.Gluc];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams[0].exams).toEqual([Exam.CBC]);
    expect(result.conditionalExams[1].exams).toEqual([Exam.HbA1C]);
  });
  it('set of preop exams for VHD + Diabetes to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams[0].exams).toMatchSnapshot();
    expect(result.conditionalExams[1].exams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for VHD + Malignancy', () => {
    result = PatientExamsNeeded([Condition.VHD, Condition.Malignancy]);
    requiredExams = [Exam.ECG, Exam.CBC, Exam.CXR];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams[0].exams).toEqual([Exam.RenPanel, Exam.PTTINR]);
  });
  it('set of preop exams for VHD + Diabetes to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams[0].exams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for Diabetes + Malignancy', () => {
    result = PatientExamsNeeded([Condition.Diabetes, Condition.Malignancy]);
    requiredExams = [Exam.ECG, Exam.RenPanel, Exam.Gluc, Exam.CBC, Exam.CXR];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams[0].exams).toEqual([Exam.HbA1C]);
    expect(result.conditionalExams[1].exams).toEqual([Exam.PTTINR]);
  });
  it('set of preop exams for Diabetes + Malignancy to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams[0].exams).toMatchSnapshot();
    expect(result.conditionalExams[1].exams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for Diabetes + Anticoagulant', () => {
    result = PatientExamsNeeded([Condition.Diabetes, Condition.Anticoagulant]);
    requiredExams = [Exam.ECG, Exam.RenPanel, Exam.Gluc, Exam.CBC];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams[0].exams).toEqual([Exam.HbA1C]);
    expect(result.conditionalExams[1].exams).toEqual([Exam.PTTINR]);
  });
  it('set of preop exams for Diabetes + Anticoagulant to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams[0].exams).toMatchSnapshot();
    expect(result.conditionalExams[1].exams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for VHD + Diabetes + Antiplatelet', () => {
    result = PatientExamsNeeded([Condition.VHD, Condition.Diabetes, Condition.Antiplatelet]);
    requiredExams = [Exam.ECG, Exam.RenPanel, Exam.Gluc, Exam.CBC];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams[0].exams).toEqual([Exam.HbA1C]);
    expect(result.conditionalExams[1].exams).toEqual([Exam.PTTINR]);
  });
  it('set of preop exams for VHD + Diabetes + Antiplatelet to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams[0].exams).toMatchSnapshot();
    expect(result.conditionalExams[1].exams).toMatchSnapshot();
  });
});

describe('Testing PatientExamsNeeded() with multiple random comorbidities', () => {
  it('returns correct set of preop exams for CAD + VHD + Hepatic + Thyroid', () => {
    result = PatientExamsNeeded([Condition.CAD, Condition.VHD,
      Condition.Hepatic, Condition.Thyroid]);
    requiredExams = [Exam.ECG, Exam.CBC, Exam.RenPanel, Exam.PTTINR, Exam.LFT, Exam.TSH];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual([]);
  });
  it('set of preop exams for CAD + VHD + Hepatic + Thyroid to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for Defib + Stroke + PulmDisease + Anemia', () => {
    result = PatientExamsNeeded([Condition.Defib, Condition.Stroke,
      Condition.PulmDisease, Condition.Anemia]);
    requiredExams = [Exam.ECG, Exam.CBC, Exam.CXR];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual([]);
  });
  it('set of preop exams for Defib + Stroke + PulmDisease + Anemia to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for AtrialFib + PulmonaryVascular + Bleeding + Digoxin', () => {
    result = PatientExamsNeeded([Condition.AtrialFib, Condition.PulmonaryVascular,
      Condition.Bleeding, Condition.Digoxin]);
    requiredExams = [Exam.ECG, Exam.GnS, Exam.CBC, Exam.PTTINR, Exam.RenPanel];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual([]);
  });
  it('set of preop exams for AtrialFib + PulmonaryVascular + Bleeding + Digoxin to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for VHD + Malignancy + Adrenal + Endocrine', () => {
    result = PatientExamsNeeded([Condition.VHD, Condition.Malignancy,
      Condition.Adrenal, Condition.Endocrine]);
    requiredExams = [Exam.ECG, Exam.CBC, Exam.CXR, Exam.RenPanel];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams[0].exams).toEqual([Exam.PTTINR]);
  });
  it('set of preop exams for VHD + Malignancy + Adrenal + Endocrine to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams[0].exams).toMatchSnapshot();
  });

  it('returns correct set of preop exams for Defib + CardiacStent + Hepatic + Anticoagulants', () => {
    result = PatientExamsNeeded([Condition.Defib, Condition.CardiacStent,
      Condition.Hepatic, Condition.Anticoagulant]);
    requiredExams = [Exam.ECG, Exam.CBC, Exam.RenPanel, Exam.PTTINR, Exam.LFT];
    expect(result.exams).toEqual(requiredExams);
    expect(result.conditionalExams).toEqual([]);
  });
  it('set of preop exams for Defib + CardiacStent + Hepatic + Anticoagulants to match snapshot', () => {
    expect(result.exams).toMatchSnapshot();
    expect(result.conditionalExams).toMatchSnapshot();
  });
});
