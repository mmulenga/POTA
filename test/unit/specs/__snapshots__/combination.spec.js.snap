// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for Diabetes + Anticoagulant to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
  "Gluc",
  "CBC",
  "PTT/INR",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for Diabetes + Anticoagulant to match snapshot 2`] = `
Array [
  "HbA1C",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for Diabetes + Malignancy to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
  "Gluc",
  "CBC",
  "CXR",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for Diabetes + Malignancy to match snapshot 2`] = `
Array [
  "HbA1C",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for Diabetes + Malignancy to match snapshot 3`] = `
Array [
  "PTT/INR",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for VHD + Diabetes + Antiplatelet to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
  "Gluc",
  "CBC",
  "PTT/INR",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for VHD + Diabetes + Antiplatelet to match snapshot 2`] = `
Array [
  "HbA1C",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for VHD + Diabetes to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
  "Gluc",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for VHD + Diabetes to match snapshot 2`] = `
Array [
  "CBC",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for VHD + Diabetes to match snapshot 3`] = `
Array [
  "HbA1C",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for VHD + Diabetes to match snapshot 4`] = `
Array [
  "ECG",
  "CBC",
  "CXR",
]
`;

exports[`Testing PatientExamsNeeded() with multiple conditional comorbidities set of preop exams for VHD + Diabetes to match snapshot 5`] = `
Array [
  "Renal Panel (Creat + Lytes)",
  "PTT/INR",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for AtrialFib + Defib to match snapshot 1`] = `
Array [
  "ECG",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for AtrialFib + Defib to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for Bleeding + Digoxin + ACEI 1`] = `
Array [
  "G&S",
  "CBC",
  "PTT/INR",
  "ECG",
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for Bleeding + Digoxin + ACEI 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for Bleeding + Digoxin + ACEI 3`] = `
Array [
  "G&S",
  "CBC",
  "PTT/INR",
  "ECG",
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for Bleeding + Digoxin + ACEI 4`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for CAD + Bleeding to match snapshot 1`] = `
Array [
  "ECG",
  "G&S",
  "CBC",
  "PTT/INR",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for CAD + Bleeding to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for CardiacStent + Adrenal to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for CardiacStent + Adrenal to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for CardiacStent + Adrenal to match snapshot 3`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for CardiacStent + Adrenal to match snapshot 4`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for CerebralDiseaese + Lithium to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for CerebralDiseaese + Lithium to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for Endocrine + Thyroid + NSAIDS 1`] = `
Array [
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for Endocrine + Thyroid + NSAIDS 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for HeartFail + Hepatic to match snapshot 1`] = `
Array [
  "ECG",
  "CBC",
  "Renal Panel (Creat + Lytes)",
  "PTT/INR",
  "LFT",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for HeartFail + Hepatic to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for PVD + Anemia to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
  "CBC",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for PVD + Anemia to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for PulmDisease + Renal + Steroid to match snapshot 1`] = `
Array [
  "ECG",
  "CBC",
  "CXR",
  "Renal Panel (Creat + Lytes)",
  "Gluc",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for PulmDisease + Renal + Steroid to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for PulmonaryVascular + Pituitary + Steroid to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
  "Gluc",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for PulmonaryVascular + Pituitary + Steroid to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for Stroke + Digoxin to match snapshot 1`] = `
Array [
  "ECG",
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for Stroke + Digoxin to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for TIA + Hepatic + Steroid to match snapshot 1`] = `
Array [
  "ECG",
  "CBC",
  "Renal Panel (Creat + Lytes)",
  "PTT/INR",
  "LFT",
  "Gluc",
]
`;

exports[`Testing PatientExamsNeeded() with multiple non-conditional comorbidities set of preop exams for TIA + Hepatic + Steroid to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for AtrialFib + PulmonaryVascular + Bleeding + Digoxin to match snapshot 1`] = `
Array [
  "ECG",
  "G&S",
  "CBC",
  "PTT/INR",
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for AtrialFib + PulmonaryVascular + Bleeding + Digoxin to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for CAD + VHD + Hepatic + Thyroid to match snapshot 1`] = `
Array [
  "ECG",
  "CBC",
  "Renal Panel (Creat + Lytes)",
  "PTT/INR",
  "LFT",
]
`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for CAD + VHD + Hepatic + Thyroid to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for Defib + CardiacStent + Hepatic + Anticoagulants to match snapshot 1`] = `
Array [
  "ECG",
  "CBC",
  "Renal Panel (Creat + Lytes)",
  "PTT/INR",
  "LFT",
]
`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for Defib + CardiacStent + Hepatic + Anticoagulants to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for Defib + Stroke + PulmDisease + Anemia to match snapshot 1`] = `
Array [
  "ECG",
  "CBC",
  "CXR",
]
`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for Defib + Stroke + PulmDisease + Anemia to match snapshot 2`] = `Array []`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for VHD + Malignancy + Adrenal + Endocrine to match snapshot 1`] = `
Array [
  "ECG",
  "CBC",
  "CXR",
  "Renal Panel (Creat + Lytes)",
]
`;

exports[`Testing PatientExamsNeeded() with multiple random comorbidities set of preop exams for VHD + Malignancy + Adrenal + Endocrine to match snapshot 2`] = `
Array [
  "PTT/INR",
]
`;
